### PR TITLE
[UXE-4335] fix: update rules engine list response phase after success creation

### PR DESCRIPTION
**Pull request by matheus-sunderhus**
### Explain what was fixed and the correct behavior.
Ensure that  the Edge application - Rules Engine - Response phase list is updated after a successfully creation .
- Fix empty search results message
- fix flaky test implementation of load-your-service-plan-service, by using vitest fake timers instead of the production code using JS Date Class.
- remove dead code inside list-template-block by removing unused slot 

### Does this PR introduce UI changes? Add a video or screenshots here.
Yes

https://github.com/user-attachments/assets/af87bba6-74d2-428f-ac05-5c4e1b6376c8

<hr />

#### Make sure your pull request fits the checklist below (when applicable):

- [x] The issue title follows the format: [ISSUE_CODE] bug: TITLE
- [x] Commits are tagged with the right word (feat, test, refactor, etc)
- [x] Application responsiveness was tested to different screen sizes
- [ ] New tests are added to prevent the same issue from happening again
- [x] UI changes are validated by a team designer
- [x] Code is formatted and linted
- [x] Tags are added to the PR

#### These changes were tested on the following browsers:
- [x] Chrome
- [ ] Edge
- [ ] Firefox
- [x] Safari

### CODE REVIEW

File: src/views/EdgeApplicationsRulesEngine/ListView.vue [hunk 1]
```diff
@@ -75,7 +74,16 @@
     }
   })
 
+  const drawerRulesEngineRef = ref('')
   const hasContentToList = ref(true)
+  const listRulesEngineRef = ref(null)
+
+  const phaseOptions = ref(['Request phase', 'Response phase'])
```
Comment (william-tome) [inline src/views/EdgeApplicationsRulesEngine/ListView.vue:81]: Não precisa ser reativo pelo que vi.

```suggestion
  const PHASE_OPTIONS = ['Request phase', 'Response phase']
```

File: src/views/EdgeApplicationsRulesEngine/ListView.vue [hunk 2]
```diff
@@ -188,6 +183,10 @@
       service: deleteRulesEngineWithDecorator
     }
   ]
+
+  watch(selectedPhase, () => {
+    listRulesEngineRef.value.reload()
+  })
```
Comment (HerbertJulio) [inline src/views/EdgeApplicationsRulesEngine/ListView.vue:189]: Sugestão: O componente `SelectButton`, que recebe a propriedade `selectedPhase`, possui eventos (`emits`) que possivelmente podem ser utilizados em vez de um `watch`. 
Recomendo analisar a possibilidade de utilizar esses eventos para melhorar a implementação: https://v3.primevue.org/selectbutton/#api.selectbutton.emits

**Review by william-tome — CHANGES_REQUESTED** [review]
Alteração simples, mas que economiza memória para o usuário.

**Review by HerbertJulio — CHANGES_REQUESTED** [review]
Waiting for change

File: src/views/EdgeApplicationsRulesEngine/ListView.vue (hunk 2 above)
Comment (matheus-sunderhus) [reply src/views/EdgeApplicationsRulesEngine/ListView.vue]: Obrigado pela dica, realmente funcionou sem o watch e foi possível remover isso da view.

**Review by HerbertJulio — APPROVED** [review]

**Review by william-tome — APPROVED** [review]

**Pull request merged**